### PR TITLE
bug 1979671: UPSTREAM <carry>: Remove pod warning annotation when workload partitioning is disabled

### PR DESCRIPTION
**Pull request by pixelsoccupied**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
When workload partitioning (currently available only for SNO) is disabled, user will not see any warning related to it. With this PR, the warning specified in the bugzilla ticket below is removed.

#### Which issue(s) this PR fixes:
Fixes **https://bugzilla.redhat.com/show_bug.cgi?id=1979671**

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by cynepco3hahue** [comment]
1. Please update commits titles and remove unneeded merge commit.
2. Please update unittests.

**Comment by pixelsoccupied** [comment]
@cynepco3hahue  can I get another lgtm? I rebased and the original one got removed

**Comment by soltysh** [comment]
/hold
we're currently bumping k8s, so this PR has to wait until that happens
/cc @damemi

**Comment by soltysh** [comment]
> /cc @soltysh can we remove the hold label ?

You'll want to observe #1087 only then we can land this.

**Comment by pixelsoccupied** [comment]
@soltysh looks like #1087 has been merged. Is it safe to remove the hold label? 

/cc @cynepco3hahue

File: openshift-kube-apiserver/admission/autoscaling/managementcpusoverride/admission.go
Comment (soltysh) [inline openshift-kube-apiserver/admission/autoscaling/managementcpusoverride/admission.go:219]: Shouldn't that error be presented when workload partitioning is enabled still? You're removing that bit entirely, not sure if that's your intention.

**Review by soltysh — APPROVED** [review]
/lgtm
/approve
/hold cancel

**Pull request merged**
